pkg/kube/translation/annotations/rewrite.go

```diff
@@ -0,0 +1,54 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package annotations
+
+import (
+	"github.com/apache/apisix-ingress-controller/pkg/log"
+	apisixv1 "github.com/apache/apisix-ingress-controller/pkg/types/apisix/v1"
+)
+
+const (
+	_rewriteTarget              = "k8s.apisix.apache.org/rewrite-target"
+	_rewriteTargetRegex         = "k8s.apisix.apache.org/rewrite-target-regex"
+	_rewriteTargetRegexTemplate = "k8s.apisix.apache.org/rewrite-target-regex-template"
+)
+
+type rewrite struct{}
+
+// NewIPRestrictionHandler creates a handler to convert
+// annotations about client ips control to APISIX ip-restrict plugin.
+func NewRewriteHandler() Handler {
+	return &rewrite{}
+}
+
+func (i *rewrite) PluginName() string {
+	return "proxy-rewrite"
+}
+
+func (i *rewrite) Handle(e Extractor) (interface{}, error) {
+	var plugin apisixv1.RewriteConfig
+	log.Errorw("handle rewrite annotations")
+	rewriteTarget := e.GetStringAnnotation(_rewriteTarget)
+	rewriteTargetRegex := e.GetStringAnnotation(_rewriteTargetRegex)
+	rewriteTemplate := e.GetStringAnnotation(_rewriteTargetRegexTemplate)
+	if rewriteTarget != "" || rewriteTargetRegex != "" || rewriteTemplate != "" {
```

Comment:
```suggestion
	if rewriteTarget != "" ||( rewriteTargetRegex != "" && rewriteTemplate != "") {
```

Reply:
Even if this check passed, we still to need verify if they are non-empty before we assign regex_uri. Because string array `[]string{"", ""}` no considered as empty.

Reply:
OK, not a big deal.